app/graphql/types/system.rb

```diff
@@ -64,6 +69,12 @@ def rules_failed(args = {})
       end
     end
 
+    def rules_failed_by_severity(args = {})
+      ::RecordLoader.for(::Host).load(object.id).then do |host|
+        host.failed_rules.reject { |rule| args[:severity].exclude?(rule.severity) }
```

Comment:
This is still causing an N+1, you need to have a second level of `CollectionLoader` that you load the `failed_rules` with.